### PR TITLE
Add CLI command to import protected embeds from WordPress.com

### PR DESCRIPTION
**Pull request by joshbetz**

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?php
```
Comment (mjangda) [inline index.php]: Should rename this file to `vip-go-wpcom-compat.php`

File: wp-cli.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	function import_protected_embeds( $args ) {
+		list( $file ) = $args;
+
+		if ( ! file_exists( $file ) ) {
+			WP_CLI::error( 'Specified file does not exist' );
+		}
+
+		global $wpdb;
+
+		$wpdb->query( 'CREATE TABLE IF NOT EXISTS `protected_embeds` (
```
Comment (mjangda) [inline wp-cli.php]: Should switch this to `dbDelta`

File: wp-cli.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+			`embed_id` varchar(64) NOT NULL COMMENT \'unique id md5 ( html )\',
+			`src` varchar(255) NOT NULL COMMENT \'source url\',
+			`embed_group_id` varchar(64) NOT NULL COMMENT \'a unique id for similar widgets\',
+			`html` mediumtext,
+			`time_added` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+			PRIMARY KEY (`id`),
+			UNIQUE KEY `embed_id` (`embed_id`)
+		) ENGINE=InnoDB AUTO_INCREMENT=0 DEFAULT CHARSET=latin1' );
+
+		$fd = fopen( $file, 'r' );
```
Comment (mjangda) [inline wp-cli.php]: Should have error handling here if the `fopen` fails.

File: wp-cli.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+		$rows = fgetcsv( $fd );
+
+		foreach ( $rows as $row ) {
+			$insert = $wpdb->insert( 'protected_embeds', $row );
+			if ( ! $insert ) {
+				WP_CLI::warning( "Could not insert embed: `{$row['id']}`" );
+			}
+		}
```
Comment (mjangda) [inline wp-cli.php]: Minor: would be nice to include a message at the end showing how many were imported vs errored.

File: wp-cli.php [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+			UNIQUE KEY `embed_id` (`embed_id`)
+		) ENGINE=InnoDB AUTO_INCREMENT=0 DEFAULT CHARSET=latin1' );
+
+		$fd = fopen( $file, 'r' );
+		$rows = fgetcsv( $fd );
+
+		foreach ( $rows as $row ) {
+			$insert = $wpdb->insert( 'protected_embeds', $row );
+			if ( ! $insert ) {
+				WP_CLI::warning( "Could not insert embed: `{$row['id']}`" );
```
Comment (mjangda) [inline wp-cli.php]: Might be helpful to include `$wpdb->last_error` in the output.

**Review by mjangda — CHANGES_REQUESTED** [review]

**Comment by joshbetz** [comment]
@mjangda thanks for reviewing. I've addressed the feedback and run through the process on a test site. So far, so good.

File: wp-cli.php [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+		$q = dbDelta( $sql );
+
+		WP_CLI::line( $q[ 'protected_embeds' ] );
+
+		$fd = fopen( $file, 'r' );
+		if ( ! $fd ) {
+			WP_CLI::error( sprintf( 'Could not open file: %s', $file ) );
+		}
+
+		$header = fgetcsv( $fd );
```
Comment (joshbetz) [inline wp-cli.php]: Maybe want to verify we have exactly the fields we expect here and probably before we go through the process of creating a new table.

Comment (mjangda) [reply wp-cli.php]: 💯 -- would definitely be a good idea.

File: wp-cli.php [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+		$sql = 'CREATE TABLE `protected_embeds` ( ' .
+			'`id` bigint(20) unsigned NOT NULL AUTO_INCREMENT, ' .
+			'`embed_id` varchar(64) NOT NULL, ' .
+			'`src` varchar(255) NOT NULL, ' .
+			'`embed_group_id` varchar(64) NOT NULL, ' .
+			'`html` mediumtext, ' .
+			'`time_added` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP, ' .
+			'PRIMARY KEY (`id`), ' .
+			'UNIQUE KEY `embed_id` (`embed_id`) ' .
+			') ENGINE=InnoDB AUTO_INCREMENT=0 DEFAULT CHARSET=latin1';
```
Comment (joshbetz) [inline wp-cli.php]: Probably need to pick the default charset here.

Comment (mjangda) [reply wp-cli.php]: Shouldn't it just match whatever charset the WP.com table uses?

Comment (joshbetz) [reply wp-cli.php]: We export to CSV on wp.com, which should convert it to utf8. Then we import from CSV using $wpdb, which should convert it to whatever charset is set in wp-config. I'm worried that using a different charset than what's set in wp-config here will cause issues if the data is ever moved.

**Pull request merged**